### PR TITLE
[HttpClient] fail properly when the server replies with HTTP/0.9

### LINKED ISSUE #37923: [HttpClient] PHP Fatal error:  Uncaught TypeError: Argument 2 passed to Symfony\Component\HttpClient\Chunk\DataChunk::__construct() must be of the type string, null given
**Symfony version(s) affected**: 5.1.3

**Description**  
Any idea why this happening? This url was in a loop and crashes the entire script.

**How to reproduce**  
```
<?php

require __DIR__ . '/vendor/autoload.php';

use Symfony\Component\HttpClient\HttpClient;

$client = HttpClient::create();
$response = $client->request('GET', 'http://123c.vn/');

```

**Additional context**  
```
PHP Fatal error:  Uncaught TypeError: Argument 2 passed to Symfony\Component\HttpClient\Chunk\DataChunk::__construct() must be of the type string, null given, called in /bug/vendor/symfony/http-client/Response/ResponseTrait.php on line 363 and defined in /bug/vendor/symfony/http-client/Chunk/DataChunk.php:26
Stack trace:
#0 /bug/vendor/symfony/http-client/Response/ResponseTrait.php(363): Symfony\Component\HttpClient\Chunk\DataChunk->__construct(0, NULL)
#1 /bug/vendor/symfony/http-client/Response/ResponseTrait.php(237): Symfony\Component\HttpClient\Response\CurlResponse::stream(Array)
#2 /bug/vendor/symfony/http-client/Response/ResponseTrait.php(300): Symfony\Component\HttpClient\Response\CurlResponse::initialize(Object(Symfony\Component\HttpClient\Response\CurlResponse))
#3 /bug/vendor/symfony/http-client/Response/CurlResponse.php(196): Symfony\Component\HttpClient\Response\CurlResponse->doDestruct()
#4 [internal  in /bug/vendor/symfony/http-client/Chunk/DataChunk.php on line 26

Fatal error: Uncaught TypeError: Argument 2 passed to Symfony\Component\HttpClient\Chunk\DataChunk::__construct() must be of the type string, null given, called in /bug/vendor/symfony/http-client/Response/ResponseTrait.php on line 363 and defined in /bug/vendor/symfony/http-client/Chunk/DataChunk.php:26
Stack trace:
#0 /bug/vendor/symfony/http-client/Response/ResponseTrait.php(363): Symfony\Component\HttpClient\Chunk\DataChunk->__construct(0, NULL)
#1 /bug/vendor/symfony/http-client/Response/ResponseTrait.php(237): Symfony\Component\HttpClient\Response\CurlResponse::stream(Array)
#2 /bug/vendor/symfony/http-client/Response/ResponseTrait.php(300): Symfony\Component\HttpClient\Response\CurlResponse::initialize(Object(Symfony\Component\HttpClient\Response\CurlResponse))
#3 /bug/vendor/symfony/http-client/Response/CurlResponse.php(196): Symfony\Component\HttpClient\Response\CurlResponse->doDestruct()
#4 [internal  in /bug/vendor/symfony/http-client/Chunk/DataChunk.php on line 26
```

Tried this on the terminal
```
curl -I http://123c.vn/
curl: (1) Received HTTP/0.9 when not allowed
```

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #37923
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by stof** [comment]
what about other client implementations ?

**Comment by stof** [comment]
and this seems to break the http push tests

File: src/Symfony/Component/HttpClient/Response/ResponseTrait.php [hunk 1]
```diff
@@ -349,8 +349,13 @@ public static function stream(iterable $responses, float $timeout = null): \Gene
                         $elapsedTimeout = 0;
 
                         if (\is_string($chunk = array_shift($multi->handlesActivity[$j]))) {
+                            if (null === $response->inflate) {
+                                $multi->handlesActivity[$j] = [null, new TransportException(sprintf('Unsupported protocol for "%s"', $response->getInfo('url')))];
```
Comment (fabpot) [inline src/Symfony/Component/HttpClient/Response/ResponseTrait.php:353]: missing dot at the end :)

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by nicolas-grekas** [comment]
> what about other client implementations ?

they already fail properly, only curl knows about HTTP/0.9

> and this seems to break the http push tests and many tests on Travis

fixed hopefuly

**Pull request merged**